File: prism/ext/jas-2.5/examples/hermite.py
#
# jython examples for jas.
# $Id: hermite.py 2448 2009-02-26 13:31:07Z kredel $
#

import sys;

from jas import Ring
from jas import Ideal
from jas import startLog
from jas import terminate

# hermite polynomial example
# H(0) = 1
# H(1) = 2 * x
# H(n) = 2 * x * H(n-1) - 2 * (n-1) * H(n-2)

r = Ring( "Z(x) L" );
print "Ring: " + str(r);
print;

# sage like: with generators for the polynomial ring
[one,x] = r.gens();

x2 = 2 * x;

N = 10;
H = [one,x2];
for n in range(2,N):
    h = x2 * H[n-1] - 2 * (n-1) * H[n-2];
    H.append( h );

for n in range(0,N):
    print "H[%s] = %s" % (n,H[n]);

print;

#sys.exit();

File: prism/ext/jas-2.5/examples/hermite.rb
#
# jruby examples for jas.
# $Id: hermite.rb 3794 2011-10-09 17:39:11Z kredel $
#

require "examples/jas"

# hermite polynomial example
# H(0) = 1
# H(1) = 2 * x
# H(n) = 2 * x * H(n-1) - 2 * (n-1) * H(n-2)

r = Ring.new( "Z(x) L" );
puts "Ring: " + str(r);
puts;

# sage like: with generators for the polynomial ring
one,x = r.gens();

x2 = 2 * x;

N = 10;
H = [one,x2];
for n in 2..N
    h = x2 * H[n-1] - 2 * (n-1) * H[n-2];
    H << h;
end

for n in 0..N
  puts "H[#{n}] = #{H[n]}";
end
puts;

#sys.exit();
